fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,187 @@
+        The type of plot to display. i.e. "bar", "line", etc
+
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection fo disaggregated metric values, along with the metric errors.
+
+    metric : str
+        The name of the metric to plot. Should match a column from the given `MetricFrame`
+
+    error_mapping : dict
+        The mapping between metrics and their corresponding metric errors.
+        
+        Note:
+            This class references columns from the `fairlearn.metrics.MetricFrame`
+            by their column (??) name.
+
+        The supported formats of errors are below:
+        +--------------------+-------------------+----------+-------------------+
+        | Error Format       |  Key_Type         | Example  | Resulting Bounds  |
+        +====================+===================+==========+===================+
+        | Symmetric Error (±)|  symmetric_error  | 0.01     | [X-0.01, X+0.01]  |
+        +--------------------+-------------------+----------+-------------------+
+        | Asymmetric Error   |  lower_error      | 0.02     | [X-0.02, X+0.03]  |
+        |                    +-------------------+----------+                   |
+        |                    |  upper_error      | 0.03     |                   |
+        +--------------------+-------------------+----------+-------------------+
+        | Bounds (symmetric  |  lower_bound      | 0.805    | [0.805, 0.815]    |
+        | or asymmetric)     +-------------------+----------+                   |
+        |                    |  upper_bound      | 0.815    |                   |
+        +--------------------+-------------------+----------+-------------------+
+
+    ax : matplotlib.axes._axes.Axes, optional
+        Custom Matplotlib axes to which this function can plot
+
+    show_plot : bool, optional
+        Whether to show the plot or not
+
+    plot_error_labels : bool, optional
+        Whether or not to plot numerical labels for the error bounds
+
+    text_precision_digits : int, optional
+        The number of digits of precision to show for error labels
+
+    text_font_size : int, optional
+        The font size to use for error labels
+
+    text_color : str, optional
+        The font color to use for error labels
+
+    text_ha : str, optional
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    matplotlib.axes._axes.Axes
+    """
+
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+    # ambiguous error metric
+    if error_bars is not None and conf_intervals is not None:
+        raise RuntimeError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+
+    if ax is None:
+        ax = plt.axes(title=title)
+    df = metric_frame.by_group
+
+    assert isinstance(metrics, list) or isinstance(metrics, str) or metrics is None, _METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics))
```

Comment:
To avoid having an AssertionError, you could combine this with the following line into a more 'written out' `if` block.